internal/services/communication/communication_service_email_domain_association_resource_test.go

```diff
@@ -0,0 +1,213 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type CommunicationServiceEmailDomainAssociationResource struct{}
+
+func TestAccCommunicationServiceEmailDomainAssociationResource_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_communication_service_email_domain_association", "test")
+	r := CommunicationServiceEmailDomainAssociationResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccCommunicationServiceEmailDomainAssociationResource_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_communication_service_email_domain_association", "test")
+	r := CommunicationServiceEmailDomainAssociationResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		{
+			Config:      r.requiresImport(data),
+			ExpectError: acceptance.RequiresImportError("azurerm_communication_service_email_domain_association"),
+		},
+	})
+}
+
+func TestAccCommunicationServiceEmailDomainAssociationResource_deleted(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_communication_service_email_domain_association", "test")
+	r := CommunicationServiceEmailDomainAssociationResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				data.CheckWithClient(r.destroy),
+			),
+			ExpectNonEmptyPlan: true,
+		},
+	})
+}
+
+func (r CommunicationServiceEmailDomainAssociationResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	exists := false
+
+	id, err := commonids.ParseCompositeResourceID(state.ID, &communicationservices.CommunicationServiceId{}, &domains.DomainId{})
+	if err != nil {
+		return &exists, fmt.Errorf("parsing ID: %w", err)
+	}
+
+	serviceClient := client.Communication.ServiceClient
+	existingCommunicationService, err := serviceClient.Get(ctx, *id.First)
+	if err != nil && !response.WasNotFound(existingCommunicationService.HttpResponse) {
+		return &exists, fmt.Errorf("checking for the presence of existing PrivateEndpoint %q: %+v", id.First, err)
+	}
+
+	if response.WasNotFound(existingCommunicationService.HttpResponse) {
+		return &exists, fmt.Errorf("PrivateEndpoint %q does not exsits", id.First)
+	}
+
+	input := existingCommunicationService
+	if input.Model != nil && input.Model.Properties != nil && input.Model.Properties.LinkedDomains != nil {
+		exists = slices.Contains(*input.Model.Properties.LinkedDomains, id.Second.ID())
+	}
+
+	return &exists, nil
```

Comment:
I think this will need to account for casing, and to be more in line with other resources, so perhaps:
```suggestion
	id, err := commonids.ParseCompositeResourceID(state.ID, &communicationservices.CommunicationServiceId{}, &domains.DomainId{})
	if err != nil {
		return pointer.To(false), fmt.Errorf("parsing ID: %w", err)
	}

	serviceClient := client.Communication.ServiceClient
	existingCommunicationService, err := serviceClient.Get(ctx, *id.First)
	if err != nil && !response.WasNotFound(existingCommunicationService.HttpResponse) {
		return pointer.To(false), fmt.Errorf("checking for the presence of existing %s: %+v", id.First, err)
	}

	if response.WasNotFound(existingCommunicationService.HttpResponse) {
		return pointer.To(false), fmt.Errorf("%s does not exist", id.First)
	}

	input := existingCommunicationService
	if input.Model != nil && input.Model.Properties != nil {
		for _, v := range pointer.From(input.Model.Properties.LinkedDomains) {
			tmpID, tmpErr := domains.ParseDomainID(v)
			if tmpErr != nil {
				return pointer.To(false), fmt.Errorf("parsing domain ID %q from LinkedDomains for %s: %+v", v, id.First, err)
			}
			if strings.EqualFold(id.Second.ID(), tmpID.ID()) {
				return pointer.To(true), nil
			}
		}
	}

	return pointer.To(false), nil
```